xgcm/test/test_autogenerate.py

```diff
@@ -0,0 +1,152 @@
+                                               {'axis': 'Y'}),
+                             'zz_inferred': (['lat', 'lon', 'z_inferred'], zz+dz,
+                                             {'axis': 'Z'})}
+                        )
+
+ds_out_right = xr.Dataset(
+                     {'somedata': (['lat', 'lon', 'z'],
+                                   a)},
+                     coords={'lon': (['lon', ], x+(dx/2.0),
+                                     {'axis': 'X',
+                                     'c_grid_axis_shift': 0.5}),
+                             'lat': (['lat', ], y+(dy/2.0),
+                                     {'axis': 'Y',
+                                     'c_grid_axis_shift': 0.5}),
+                             'z': (['z', ], z+(dz/2.0),
+                                   {'axis': 'Z',
+                                   'c_grid_axis_shift': 0.5}),
+                             'llon': (['lat', 'lon'], xx+(dx/2.0),
+                                      {'axis': 'X',
+                                      'c_grid_axis_shift': 0.5}),
+                             'llat': (['lat', 'lon'], yy+(dy/2.0),
+                                      {'axis': 'Y',
+                                      'c_grid_axis_shift': 0.5}),
+                             'zz': (['lat', 'lon', 'z'], zz+(dx/2.0),
+                                    {'axis': 'Z',
+                                    'c_grid_axis_shift': 0.5}),
+                             'lon_inferred': (['lon_inferred', ], x,
+                                              {'axis': 'X'}),
+                             'lat_inferred': (['lat_inferred', ], y,
+                                              {'axis': 'Y'}),
+                             'z_inferred': (['z_inferred', ], z,
+                                            {'axis': 'Z'}),
+                             'llon_inferred': (['lat', 'lon_inferred', ], xx,
+                                               {'axis': 'X'}),
+                             'llat_inferred': (['lat_inferred', 'lon'], yy,
+                                               {'axis': 'Y'}),
+                             'zz_inferred': (['lat', 'lon', 'z_inferred'], zz,
+                                             {'axis': 'Z'})}
+                        )
+
+
+def test_autogenerate_ds():
+    axes = {'X': 'lon', 'Y': 'lat', 'Z': 'z'}
+    coord_axes = {'X': 'llon', 'Y': 'llat', 'Z': 'zz'}
+    center = autogenerate_ds(ds_original, axes=axes,
+                             position='center', coord_axes=coord_axes)
+
+    left = autogenerate_ds(ds_original, axes=axes,
+                           position='left', coord_axes=coord_axes)
+
+    right = autogenerate_ds(ds_original, axes=axes,
+                            position='right', coord_axes=coord_axes)
+
+    xr.testing.assert_identical(center, ds_out_center)
+    xr.testing.assert_identical(left, ds_out_left)
+    xr.testing.assert_identical(right, ds_out_right)
```

Comment:
I really like your test!

It is, however, a pretty high level test. It might be valuable to break down your new module into smaller units and write some unit tests.